### PR TITLE
EXSWCPHIPT-116 - Adding tests for hipSet/GetDevice and hipDeviceGet

### PR DESCRIPTION
**Pull request by jatinx**
Add multi threaded tests for hipSet/GetDevice
Add missing unit tests

### CODE REVIEW

**Review by chrispaquot — DISMISSED** [review]

**Comment by FMarno** [comment]
The negative tests for hipsetdevice are missing.

File: tests/catch/unit/device/hipSetGetDevice.cc [hunk 1]
```diff
@@ -24,20 +24,81 @@
...
+  HIP_CHECK_ERROR(hipGetDevice(nullptr), hipErrorInvalidValue);
+}
+
+TEST_CASE("Unit_hipDeviceGet_Negative") {
+  SECTION("Nullptr as handle") { HIP_CHECK_ERROR(hipDeviceGet(nullptr, 0), hipErrorInvalidValue); }
+
+  SECTION("Out of bound ordial - positive") {
+    hipDevice_t device{};
+    auto totalDevices = HipTest::getDeviceCount();
+    HIP_CHECK_ERROR(hipDeviceGet(&device, totalDevices + 1), hipErrorInvalidDevice);
```
Comment (FMarno) [inline tests/catch/unit/device/hipSetGetDevice.cc]:
```suggestion
    HIP_CHECK_ERROR(hipDeviceGet(&device, totalDevices), hipErrorInvalidDevice);
```
I think it is worth while checking the off-by-one error here.

Comment (jatinx) [reply tests/catch/unit/device/hipSetGetDevice.cc]: Done

**Review by FMarno — DISMISSED** [review]
LGTM!

**Review by chrispaquot — DISMISSED** [review]

**Comment by jatinx** [comment]
Depends on #2664

**Review by chrispaquot — DISMISSED** [review]

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**